core/cas-server-core-events-api/src/main/java/org/apereo/cas/support/events/listener/DefaultCasEventListenerImpl.java

```diff
@@ -0,0 +1,121 @@
+package org.apereo.cas.support.events.listener;
+
+import org.apereo.cas.support.events.AbstractCasEvent;
+import org.apereo.cas.support.events.CasEventRepository;
+import org.apereo.cas.support.events.authentication.CasAuthenticationPolicyFailureEvent;
+import org.apereo.cas.support.events.authentication.CasAuthenticationTransactionFailureEvent;
+import org.apereo.cas.support.events.authentication.adaptive.CasRiskyAuthenticationDetectedEvent;
+import org.apereo.cas.support.events.dao.CasEvent;
+import org.apereo.cas.support.events.ticket.CasTicketGrantingTicketCreatedEvent;
+import org.apereo.cas.support.events.ticket.CasTicketGrantingTicketDestroyedEvent;
+import org.apereo.cas.util.DateTimeUtils;
+import org.apereo.cas.util.HttpRequestUtils;
+import org.apereo.cas.util.serialization.MessageSanitizationUtils;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apereo.inspektr.common.web.ClientInfoHolder;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+
+import java.time.Instant;
+
+/**
+ * This is {@link DefaultCasEventListenerImpl} that attempts to consume CAS events
+ * upon various authentication events. Event data is persisted into a repository
+ * via {@link CasEventRepository}.
+ *
+ * @author Misagh Moayyed
+ * @since 5.0.0
+ */
+@RequiredArgsConstructor
+@Getter
+@Slf4j
+public class DefaultCasEventListenerImpl implements DefaultCasEventListener {
```

Comment:
Please avoid the name "Impl". This is a frowned-upon practice. You probably could opt for DefaultCasAuthenticationEventListener, or something that reflects the operations inside this listener.